spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/KubernetesDiscoveryClientAutoConfiguration.java

```diff
@@ -11,6 +11,7 @@
 import org.springframework.context.annotation.Primary;
 
 @Configuration
+@ConditionalOnProperty(value = "spring.cloud.kubernetes.discovery.enabled", matchIfMissing = true)
```

Comment:
Should also add `spring.cloud.kubernetes.enabled` condition as well, in case one decides to globally disable Spring Cloud Kubernetes altogether.